packages/rpc-provider/src/substrate-connect/index.ts

```diff
@@ -154,9 +154,17 @@ export class ScProvider implements ProviderInterface {
       callback?.(decodedResponse);
     };
 
-    const addChain = this.#wellKnownChains.has(this.#spec as ScType.WellKnownChain)
-      ? client.addWellKnownChain
-      : client.addChain;
+    const addChain = this.#sharedSandbox
+      ? (async (...args) => {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const source = this.#sharedSandbox!;
+
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return (await source.#chain)!.addChain(...args);
```

Comment:
`eslint-disable` because `this.#sharedSandbox.connect` is invoked on line [113](https://github.com/polkadot-js/api/pull/5776/files#diff-66ebd813ed4a22dc4419b70f5b5771ac5f597162674c041aad2ecec74717fd0dR113) and `.connect` creates the a chain instance on `.#chain` on line [169](https://github.com/polkadot-js/api/pull/5776/files#diff-66ebd813ed4a22dc4419b70f5b5771ac5f597162674c041aad2ecec74717fd0dR169)

Note: all these changes are inside the `.connect` method.